osbrain/tests/test_agent_async_requests.py

```diff
@@ -139,63 +124,73 @@ def test_unknown(nsproxy, server_client_late_reply_return):
 
     # Manually remove the pending request before it is received
     client.set_attr(_pending_requests={})
-    time.sleep(1.1)
-    assert server.get_attr('received') == ['foo']
-    assert client.get_attr('received') == []
+    server.unsafe.set_attr(blocked=False)
+    assert wait_agent_attr(server, value=['foo'], timeout=.1)
+    assert not wait_agent_attr(client, length=1, timeout=.5)
     assert logger_received(logger,
                            log_name='log_history_warning',
                            message='Received response for an unknown request!')
 
 
-def test_wait(nsproxy, server_client_late_reply_delay):
+def test_wait_in_time(nsproxy):
     """
     Asynchronous request-reply pattern maximum wait.
     """
-    server, client = server_client_late_reply_delay
+    client, server = async_client_server()
 
     logger = run_logger('logger')
     client.set_logger(logger)
     sync_agent_logger(client, logger)
 
     # Response received in time
-    client.send('async', 1, wait=2)
-    time.sleep(1.1)
-    assert server.get_attr('received') == [1]
-    assert client.get_attr('received') == ['x1']
+    client.send('async', '1', wait=2)
```

Comment:
Just for consistency with other tests should we send `'foo'` instead of `'1'`? A number was previously sent to set the server blocking duration, but this is no longer the case.

If you agree, change it in the tests bellow as well. :wink: